samuroi/util/mask_generator/create_masks.py

```diff
@@ -0,0 +1,196 @@
+        self.roi_mask_list = self.create_roi_masks()
+
+    def create_roi_masks(self):
+        roi_mask_list = []
+        for i in range(np.max(self.segmentation_labels)):
+            idx = np.where(self.segmentation_labels*self.putative_nuclei_image == i+1)
+            x = np.vstack((idx[0], idx[1]))
+            m = np.reshape(x.T, (len(idx[0]), 2))
+            roi_mask_list.append(m)
+
+        return roi_mask_list
+
+
+def get_blob_centers(segmentation_layer, iterations=3):
+    """
+    Determine the centers of each object in an image
+
+    ::param segmentation_layer: NxM ndarray image mask of all target objects
+    ::param iterations: threshold for removal of small non-target objects
+    ::returns com: a np ndarray of x,y coordinates for the center of each target object
+
+    """
+
+    segmentation_layer = ndimage.binary_opening(segmentation_layer, iterations=iterations)  # remove small objects
+    labels, label_number = ndimage.measurements.label(segmentation_layer)  # label remaining blobs
+
+    com = np.zeros((label_number, 2))
+
+    for i in range(label_number):
+        idx = np.where(labels == i+1)  # calculate the center of mass for each blob
+        com[i, 1] = np.mean(idx[1].astype(float))  # must be float
+        com[i, 0] = np.mean(idx[0].astype(float))  # must be float
+
+    return com
+
+
+def remove_small_blobs(segmentation_layer, com):
+    """
+    removes non-overlapping pixel-islands and cell centres (coms)
+    ::param segmentation_layer: NxM ndarray image mask of all target objects
+    ::param com: a np ndarray of x,y coordinates for the center of each target object
+    ::return updated_labels:
+    """
+    labels, label_number = ndimage.label(segmentation_layer)  # label all pixel islands
+    updated_labels = np.zeros_like(labels)
+    coms_to_keep = np.zeros(len(com))  # holder
+
+    for i in range(label_number):
+        idx = np.where(labels == i+1)
+        for j, c in enumerate(com.astype(int)):
+            if labels[c[0], c[1]] == i+1:  # if the com is in the blob
+                updated_labels[idx] = 1  # add the blob
+                coms_to_keep[j] = 1  # add the com
+
+    com_index = np.where(coms_to_keep == 1)
+    updated_com = com[com_index]
+
+    return updated_labels, updated_com
+
+
+def calculate_distance(image, com):
+    """
+    ::param image: a binarised image to be segmented
+    ::param com: the centres that will define the maxima of the watershed segmentation
+    """
+    # random walk segmentation of 2D image-mask array
+    distance = ndimage.distance_transform_edt(np.abs(image-1))
+    local_maxi = np.zeros_like(image)
+
+    for c in com:
+        local_maxi[int(c[0]), int(c[1])] = 1
+
+    markers = ndimage.label(local_maxi)[0]
+    segmentation_labels = segmentation.random_walker(distance, markers, beta=60)
+
+    return segmentation_labels
+
+
+def blob_labels(com, blob_image):
```

Comment:
all other functions take the parameters image and com in different order. please stay consistent.